astra/src/main/java/com/slack/astra/graphApi/Edge.java

```diff
@@ -0,0 +1,39 @@
+package com.slack.astra.graphApi;
+
+// Represents a directed connection between two Nodes.
+public final class Edge {
+  private final String sourceNodeId, targetNodeId;
+
+  Edge(String sourceNodeId, String targetNodeId) {
+    if (sourceNodeId == null) throw new NullPointerException("sourceNodeId == null");
+    if (targetNodeId == null) throw new NullPointerException("targetNodeId == null");
```

Comment:
This would match other places in the repo better. It changes the exception type to IllegalArgumentException though.
```suggestion
    Preconditions.checkNotNull(sourceNodeId);
    Preconditions.checkNotNull(targetNodeId);
```

Reply:
Looks like it throws a null pointer 
```
   public static <T> T checkNotNull(@Nullable T reference, @Nullable Object errorMessage) {
        if (reference == null) {
            throw new NullPointerException(Platform.stringValueOf(errorMessage));
        } else {
            return reference;
        }
    }
```